### ISSUE TITLE
iPhone not detected

### ISSUE THREAD

**Issue by SaltyKisses** [davesc63/GeoPort#46]
I have a box fresh iPhone 12. Purchased for use with this app. iOS version is 17.4.1
Using the Apple USB-C to Lightning cable supplied with the device.

PC1 - Windows 10, GeoPort cannot detect the iPhone although Windows can
PC 2 - Windows 11, same issue.
Can use Windows Explorer to connect to the iPhone but it is not detected in the GeoPort webUI.

Have confirmed Windows firewall settings and even disabled the firewall temporarily as a test. 

Console window log from startup shows no errors regarding inability to connect to anything.

I've looked in the FAQ and other issues here but not seeing anyone who fails at the first step.

Hoping this is something terribly basic.

2024-05-14 20:21:04 Peter_E14_Ryzen pyi_splash[20700] INFO A connection to the splash screen was successfully established.
2024-05-14 20:21:04 Peter_E14_Ryzen GeoPort[20700] INFO clear splash
2024-05-14 20:21:04 Peter_E14_Ryzen pyuac[20700] INFO isUserAdmin - CheckTokenMembership returned: True
2024-05-14 20:21:06 Peter_E14_Ryzen GeoPort[20700] INFO Serving: http://localhost:54321
 * Serving Flask app 'main-win'
 * Debug mode: off
2024-05-14 20:21:10 Peter_E14_Ryzen GeoPort[20700] WARNING Failed to retrieve country name using locale. Using IP geolocation service.
2024-05-14 20:21:10 Peter_E14_Ryzen GeoPort[20700] INFO Country: Australia
2024-05-14 20:21:10 Peter_E14_Ryzen GeoPort[20700] INFO Current platform: Windows
2024-05-14 20:21:10 Peter_E14_Ryzen GeoPort[20700] INFO App Version = 2.3.3
2024-05-14 20:21:10 Peter_E14_Ryzen GeoPort[20700] INFO base dir =  C:\Users\choic\AppData\Local\Temp\_MEI62282
2024-05-14 20:21:10 Peter_E14_Ryzen GeoPort[20700] INFO GitHub Version = 2.3.3
2024-05-14 20:25:22 Peter_E14_Ryzen GeoPort[20700] WARNING Failed to retrieve country name using locale. Using IP geolocation service.
2024-05-14 20:25:22 Peter_E14_Ryzen GeoPort[20700] INFO Country: Australia
2024-05-14 20:25:22 Peter_E14_Ryzen GeoPort[20700] INFO Current platform: Windows
2024-05-14 20:25:22 Peter_E14_Ryzen GeoPort[20700] INFO App Version = 2.3.3
2024-05-14 20:25:22 Peter_E14_Ryzen GeoPort[20700] INFO base dir =  C:\Users\choic\AppData\Local\Temp\_MEI62282
2024-05-14 20:25:22 Peter_E14_Ryzen GeoPort[20700] INFO GitHub Version = 2.3.3

**Comment by davesc63** [comment]
Your windows machines don’t have iTunes installed

**Comment by davesc63** [comment]
Install iTunes and make sure your machines are listening on port 27015. 

Please report back. I’m pretty sure you’ll be enjoying GeoPort once that is done !

**Comment by SaltyKisses** [comment]
Yes I figured that I needed iTunes. 
It's not in the FAQ but at least now there's a GitHub issue for people to find if like me, they download from a direct link rather than the homepage!

So
iTunes added, Windows rebooted. iPhone is now detected. Developer mode enabled on the iPhone

New issue - The simulate location button doesn't work. Clicking on it doesn't bring up the little tooltip shown in the documentation.

Would you prefer to close this issue and start another for 'Simulate Location button not working'?

2024-05-14 21:01:12 Peter_E14_Ryzen GeoPort[15912] INFO iOS 17+ detected
2024-05-14 21:01:12 Peter_E14_Ryzen GeoPort[15912] INFO iOS Version: 17.4.1
2024-05-14 21:01:12 Peter_E14_Ryzen GeoPort[15912] INFO Driver is not required
2024-05-14 21:01:12 Peter_E14_Ryzen GeoPort[15912] INFO Create Lockdown <PlistUsbmuxLockdownClient ID:00008101-000E11313ED2001E VERSION:17.4.1 TYPE:iPhone13,2 PAIRED:True>
2024-05-14 21:01:12 Peter_E14_Ryzen GeoPort[15912] WARNING Start USB TCP tunnel
2024-05-14 21:01:51 Peter_E14_Ryzen GeoPort[15912] ERROR RSD Data is None, Perhaps the tunnel isn't established
2024-05-14 21:01:51 Peter_E14_Ryzen GeoPort[15912] INFO Device Connection Map: {'00008101-000E11313ED2001E': {'USB': {'host': None, 'port': None}}}
2024-05-14 21:01:51 Peter_E14_Ryzen GeoPort[15912] WARNING Connect Device function completed
2024-05-14 21:09:25 Peter_E14_Ryzen GeoPort[15912] INFO RSD in udid mapping is: {'host': None, 'port': None}
2024-05-14 21:09:25 Peter_E14_Ryzen GeoPort[15912] INFO RSD already created. Reusing connection
2024-05-14 21:09:25 Peter_E14_Ryzen GeoPort[15912] INFO RSD Data: {'host': None, 'port': None}
2024-05-14 21:09:25 Peter_E14_Ryzen GeoPort[15912] ERROR Error setting location: [Errno 11001] getaddrinfo failed

**Comment by SaltyKisses** [comment]
Update. The Simulate Location button does now give a 'location set successfully' tool-tip but the log suggests otherwise.

2024-05-14 21:26:34 Peter_E14_Ryzen GeoPort[15912] ERROR Error setting location: [Errno 11001] getaddrinfo failed  

Maps and the 7-Eleven app show the true location not the simulated location so I'm thinking that something is wrong with the location setting part.

**Comment by SaltyKisses** [comment]
Yet another reboot - as a speculator - and the location can now be set.

**Comment by davesc63** [comment]
Thank you. I love it when someone can help troubleshoot!

Kill off GeoPort. Try again. This one is odd. 
If we can detect your device via usbmux. You’re on 17.4.1, so we can use the newer, better method to connect. But your log shows no rsd data. This is not expected. 

As someone else has mentioned. They ran GeoPort from a one drive synced location and that caused issues. 

We should be creating an rsd port within seconds

**Comment by SaltyKisses** [comment]
In my old job as a professional services consultant for a contact centre software manufacturer, troubleshooting this sort of thing was my bread & butter.

Last night I got the location set but always Error 01 in that fuel application.

I tried disconnecting GeoPort before the lock and swapping from home Wi-Fi to phone hotspot at the critical moments without success.

I've spent much of today with Android Emulator - my old method.
Not having any luck there.
Different VPN and no VPN.
Using different states.
Including HMA to hide the rooted apps.
Changed the device IDs.

Everything ends in the same 01 Error.
I think the most frustrating part for me with GeoPort is that I am a total n00b at iPhones.
They are so unintuitve to me after being brought up on Ericsson, Nokia, Windows Phone, Samsung etc.
I have to keep googling for basic things which would be easy if only iPhones had buttons!
Thanks for your help.

Peter
________________________________
